### PR TITLE
Add mixed no_else_raise/return testing

### PR DESCRIPTION
**Pull request by jrobertson98atx**
## Steps

- [x] Write a good description on what the PR does.

## Description

This PR adds mixed no-else-return and no-else-raise testing as suggested in issue #2558.  There are no functional changes.

## Related Issue

Testing for issue #2558.

### CODE REVIEW

**Comment by jrobertson98atx** [comment]
Hi @PCManticore - 

I'm not able to reproduce the failure for this PR.  I'm not sure how the changes in this one could have caused a failure though given they only changed the testing criteria for a couple of tests.

Do you see anything obviously wrong with this submission that would cause failures?  Any insight would be greatly appreciated.

Thanks,
Jim

**Comment by PCManticore** [comment]
Hey @jrobertson98atx This is most likely caused by the recent changes in astroid. You might not be able to reproduce them locally if you use an older version of astroid. If you use `tox`, doing `tox --recreate` should do it. Don't worry about this failure though, it's most likely unrelated to your changes.

File: pylint/test/functional/no_else_raise.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def foo2(x, y, w, z):
         raise Exception(w)
     else:
         c = 3
-        raise Exception(z)
+        return z
```
Comment (PCManticore) [inline pylint/test/functional/no_else_raise.py]: I think we should not modify this here, unless this was a bug which was fixed by this PR (but does not seem to be the case here). If we need to add mixed testing, we should add separate test case.

**Comment by jrobertson98atx** [comment]
> Hey @jrobertson98atx I don't think we should modify the current tests but instead we should add separate tests for testing mixed returns.

Ok. I'll cancel this one and add new testing.  Thanks.

**Pull request closed without merging**